.github/workflows/ci.yml

```diff
@@ -269,18 +269,104 @@ jobs:
-     
+
       - name: Test driver build on linux 5.x
         run: |
           driverkit docker --kernelrelease 5.19.12-arch1-1 --target arch --output-module /tmp/libs.ko --output-probe=/tmp/libs.o --driverversion $GITHUB_SHA --loglevel debug  --kernelurls https://download.falco.org/fixtures/libs/linux-headers-5.19.12.arch1-1-x86_64.pkg.tar.zst
+
+  run-e2e-tests-amd64:
+    name: run-e2e-tests-amd64
+    strategy:
+      matrix:
+        name: [system_deps, bundled_deps, system_deps_w_chisels]
+        include:
+          - name: system_deps
+            cmake_opts: -DBUILD_BPF=On -DUSE_BUNDLED_DEPS=False
+          - name: bundled_deps
+            cmake_opts: -DBUILD_BPF=On -DUSE_BUNDLED_DEPS=True
+          - name: system_deps_w_chisels
+            cmake_opts: -DBUILD_BPF=On -DUSE_BUNDLED_DEPS=False -DWITH_CHISEL=True
+    runs-on: ubuntu-latest
+    container:
+      image: debian:buster
+      volumes:
+        - /tmp/report/:/tmp/report/
+        - /usr/src:/usr/src
+        - /lib/modules:/lib/modules
+        - /usr/include/bpf:/usr/include/bpf
+        - /dev:/dev
+    steps:
+      - name: Install deps ⛓️
+        run: |
+          apt update && apt install -y --no-install-recommends \
+            ca-certificates \
+            cmake \
+            build-essential \
+            git \
+            clang \
+            llvm \
+            pkg-config \
+            autoconf \
+            automake \
+            libtool \
+            libelf-dev \
+            wget \
+            libb64-dev \
+            libc-ares-dev \
+            libcurl4-openssl-dev \
+            libssl-dev \
+            libtbb-dev \
+            libjq-dev \
+            libjsoncpp-dev \
+            libgrpc++-dev \
+            protobuf-compiler-grpc \
+            libgtest-dev \
+            libprotobuf-dev \
+            liblua5.1-dev \
+            curl \
+            gnupg \
+            lsb-release
+
+          # install docker-ce-cli, required for e2e tests
+          mkdir -p /etc/apt/keyrings
+          curl -fsSL https://download.docker.com/linux/debian/gpg | gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+          echo \
+            "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/debian \
+            $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null
+          apt-get update
+          apt-get install -y docker-ce-cli
+
+      - name: Checkout Libs ⤵️
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install deps ⛓️
+        run: |
+          .github/install-deps.sh
+
+      - name: Build and test 🏗️🧪
+        run: |
+          mkdir -p build
+          cd build && cmake ${{ matrix.cmake_opts }} ../
+          KERNELDIR=/lib/modules/$(uname -r)/build make -j4
+          make  e2e-tests
+
+      - name: Archive test reports
+        uses: actions/upload-artifact@v3
+        if: always()
```

Comment:
Is this needed to upload artifact even when e2e tests fail?

Reply:
No, I completely forgot to change this, sorry 😅. I'll change it to only upload on fail.